Runner/utils/functestlib.sh

```diff
@@ -132,13 +132,21 @@ unload_kernel_module() {
 # --- Dependency check ---
 check_dependencies() {
     missing=0
+    missing_cmds=""
     for cmd in "$@"; do
         if ! command -v "$cmd" >/dev/null 2>&1; then
-            log_error "Required command '$cmd' not found in PATH."
+            log_warn "Required command '$cmd' not found in PATH."
             missing=1
+            missing_cmds="$missing_cmds $cmd"
         fi
     done
-    [ "$missing" -ne 0 ] && exit 1
+    if [ "$missing" -ne 0 ]; then
+        # Use existing TESTNAME variable if present, fallback if not
+        testname="${TESTNAME:-UnknownTest}"
+        log_skip "$testname SKIP: missing dependencies:$missing_cmds"
+        echo "$testname SKIP" > "./$testname.res"
+        exit 0
```

Comment:
`exit 1` ?

Reply:
that's correct as far as I can tell. In the LAVA test plan here we use `|| true` to avoid this problem. I'm fine leaving it like it is.